src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -1,715 +1,29 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
-
-using System;
-using System.Collections.Concurrent;
+﻿using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Linq;
-using System.Runtime.Versioning;
-using System.Threading;
-using System.Threading.Tasks;
+using System.Web;
 using System.Web.Configuration;
-using NuGet.Server.Logging;
 
 namespace NuGet.Server.Infrastructure
 {
-    /// <summary>
-    /// ServerPackageRepository represents a folder of nupkgs on disk. All packages are cached during the first request in order
-    /// to correctly determine attributes such as IsAbsoluteLatestVersion. Adding, removing, or making changes to packages on disk 
-    /// will clear the cache.
-    /// </summary>
-    public class ServerPackageRepository
-        : PackageRepositoryBase, IServerPackageRepository, IPackageLookup, IDisposable
+	public class ServerPackageRepository : ServerPackageRepositoryCore
```

Comment:
Is this just to avoid system.web dependencies in the core project? I would avoid deriving a type, and just create a factory if possible.

Also don't name classes with Core suffix, use meaningful names instead